bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschHttpClient.java

```diff
@@ -111,7 +111,32 @@ public String getPairingUrl() {
      * @return Bosch SHC URL for passed endpoint
      */
     public String getBoschShcUrl(String endpoint) {
-        return String.format("https://%s:8444/%s", this.ipAddress, endpoint);
+        String url = String.format("https://%s:8444/%s", this.ipAddress, endpoint);
+        return escapeURL(url);
+    }
+
+    /**
+     * Performs specific URL escaping required for certain Bosch SHC URLs.
+     * <p>
+     * In particular, hash characters in child device IDs must be escaped with <code>%23</code>.
+     * <p>
+     * Invalid example:
+     * 
+     * <pre>
+     * https://host:port/devices/hdm:ZigBee:70ac08fffe5294ea#3/services/PowerSwitch/state
+     * </pre>
+     * 
+     * Valid example:
+     * 
+     * <pre>
+     * https://host:port/devices/hdm:ZigBee:70ac08fffe5294ea%233/services/PowerSwitch/state
+     * </pre>
+     * 
+     * @param url the URL to be escaped
+     * @return the escaped URL
+     */
+    private String escapeURL(String url) {
+        return url.replace("#", "%23");
```

Comment:
This does not seem specific to the added support for light/shutter. YDid you discover some bug?
Adding such fix is great, but for future reference/regression etc it is good to name this in the PR start post.

Reply:
This is indeed specific to the new device. The Light/Shutter Control II is the first device supported in our binding that exposes logical parent and child devices. The parent device IDs look like `hdm:ZigBee:70ac08fffe5294ea` and the child device IDs look like `hdm:ZigBee:70ac08fffe5294ea#3` (note the `#3` in the end).

These device IDs eventually become part of a REST URL like `https://host:port/devices/hdm:ZigBee:70ac08fffe5294ea#3/services/PowerSwitch/state`. But in the URL `#3` has to be escaped with `%23`.

Reply:
Ok, thanks for the clarification.